pkg/cloud/rgraph/ez/ez.go

```diff
@@ -0,0 +1,146 @@
+/*
+Copyright 2023 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package ez is a utility to create complex resource graphs for testing from a
+// concise description by use of naming conventions and default values.
+//
+// This package should only be used for testing. There is little error handling;
+// most errors will result in a panic() to reduce the verbosity of the code.
+package ez
+
+import (
+	"fmt"
+
+	"github.com/GoogleCloudPlatform/k8s-cloud-provider/pkg/cloud"
+	"github.com/GoogleCloudPlatform/k8s-cloud-provider/pkg/cloud/meta"
+	"github.com/GoogleCloudPlatform/k8s-cloud-provider/pkg/cloud/rgraph"
+)
+
+type GraphOption int
+
+const (
+	// PanicOnAccessErr causes panic() if Accces() returns an error. Otherwise,
+	// the error is ignored.
+	PanicOnAccessErr = 1 << iota
+)
+
+func panicf(s string, args ...any) { panic(fmt.Sprintf(s, args...)) }
+
+type Graph struct {
+	Nodes   []Node
+	Project string
```

Comment:
Why is Project an attribute of both `Graph` and `Node`? (I think it makes sense on Node if/when we need to support cross-project graphs, but I'm not sure why we need this in both places and how we should interpret conflicting values).

Reply:
It's to make the common case more terse:

```
		ezg := ez.Graph{
      Project: "proj-1",
			Nodes: []ez.Node{
				{Name: "addr1", Project: "proj-2"},
				{Name: "fr1", Refs: []ez.Ref{{Field: "IPAddress", To: "addr1"}, {Field: "Target", To: "thp"}}},
				{Name: "thp", Refs: []ez.Ref{{Field: "UrlMap", To: "um"}}},
				{Name: "um", Refs: []ez.Ref{{Field: "DefaultService", To: "bs"}}},
				{Name: "bs", Refs: []ez.Ref{{Field: "Backends.Group", To: "us-central1-b/neg1"}, {Field: "Healthchecks", To: "hc"}}},
				{Name: "hc"},
				{Name: "neg1", Zone: "us-central1-b"},

				{Name: "fr2", Options: ez.DoesNotExist},
				{Name: "fr3", Options: ez.DoesNotExist},
			},
		}
```

Reply:
So this is essentially a default for Nodes when they're unspecified? If so, maybe a comment would help explain that for someone new to this.

Reply:
good point, adding.